test/test_helper.rb

```diff
@@ -33,6 +33,31 @@ def with_preview_paths(new_value)
   Rails.application.config.view_component.preview_paths = old_value
 end
 
+# Accepts any config, so we dont need to create one function for each attribute
+#
+# @param configs [Array<String>] Use 'parameter = value'
+# @yield Test code to run
+# @return [void]
+# rubocop:disable Security/Eval
+def with_custom_config(*configs)
+  old_config = configs.map do |config|
+    config_name = config.split("=").first.strip
+    val = eval(config_name)
+    val = val.is_a?(Symbol) ? ":#{val}" : val
+
+    "#{config_name} = #{val}"
+  end
+
+  apply_config = ->(c) { c.each { |ee| eval(ee) } }
+
+  begin
+    apply_config.call(configs)
+    yield
+  ensure
+    apply_config.call(old_config)
+  end
+end
```

Comment:
I understand the motivation here but the use of eval is scary and the assumption that each config is only set with an `=` seems like a bit of a stretch. This would be just as simple and more direct to just have a helper like 


```suggestion
def with_locale(locale, default_locale = I18n.default_locale)
  old_locale = I18n.locale
  old_default_locale = I18n.default_locale
  
  begin
    I18n.locale = locale
    I18n.default_locale = default_locale
    yield
  ensure
     I18n.locale = old_locale
     I18n.default_locale = old_default_locale
  end
end
```

Reply:
Yep, I know it's scary... but still, it's just for tests. My idea was to use this function to replace all those other functions in test_helper.rb (with_preview_route, with_preview_controller, with_custom_component_path, with_default_preview_layout, with_render_monkey_patch_config...).

I also think its helps in readability (as the configuration is close to the test itself). But I can remove it, no problem.

Reply:
Sure but those other test helpers have other things they need like reloading. Also it's somewhat drier to use the explicit helper methods. IMO its much cleaner and declarative to see:

```ruby
with_preview_controller("MyPreviewController") do
```

than
```ruby
with_custom_config("Rails.application.config.view_component.preview_controller = 'MyPreviewController'")
```

It's less for me to read which aids in the comprehension of the code under test. 

Lastly its much more maintainable. If the setting for `preview_controller` changed for any reason with the explicit test_helper we'd only need to update one location. The `with_custom_config` approach requires you to change it everywhere its used.

Reply:
I had done this change to work with reloading and changed all tests locally, it would be the next PR... :sweat_smile: 

One last comment, it also would help to join multiple configurations too...

``ruby
with_custom_config(
  "Rails.application.config.view_component.preview_controller = 'MyPreviewController'",
  "I18n.locale = ':en'"
)
``

But no problem, I'll remove it.